src/walletConnect/screens/RequestContent.tsx

```diff
@@ -28,6 +30,36 @@ interface Props {
 
 const DAPP_IMAGE_SIZE = 60
 
+export const useDappMetadata = (metadata?: IClientMeta | CoreTypes.Metadata | null) => {
+  const activeDapp = useSelector(activeDappSelector)
+
+  if (!metadata) {
+    // should never happen
+    return {
+      url: '',
+      dappName: '',
+      dappImageUrl: '',
+    }
+  }
+
+  const { url, name, icons } = metadata
+  const dappHostname = url ? new URL(url).hostname : ''
```

Comment:
Sorry this can still crash with `new URL('/en-US/docs')` for instance.
We have to use a try/catch block.

Reply:
ah, third time's the charm...i read the docs and updated an existing hook that uses this logic